### ISSUE TITLE
[Critical] Role: The element's role is missing or incorrect

### ISSUE THREAD

**Issue by ntallman** [APTrust/dart#574]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87611 |
| **Checkpoint** | 4.1 Compatible › Custom Controls (4.1.2.b) |
| **Impact** | Critical |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 01 Settings Workflow |
| **URL** | <https://aptrust.org/documentation-page/dart-digital-archivists-resource-tool/> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Date Reported** | 05/06/26 at 5:03 pm |
| **Issue Status** | Open |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

The element's role is missing or is not appropriate for the element's function.
Pattern: For the confirmation message bar close button is coded as "same page link" which can be confusing. See Settings User Experience video at 3:25.
To reproduce, select Remove Repositories from the Settings menu, then select New - enter Name and URL, then select Save.  A confirmation message will appear (Remote repo settings saved.) with a close button (or dismiss) the alert banner.

## Recommendation

Rule : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

How To Fix: 
Fix this issue by only using appropriate, valid HTML <button> or ARIA roles per the ARIA Recommendation document and the ARIA Authoring Practices document (see References below).

Reference: 
Links vs Buttons: https://www.a11y-collective.com/blog/button-vs-link/

Background: 
Every user interface control must have a role to convey what type of control it is for screen reader and other assistive technology users. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) using ARIA as well as expected keyboard interactions.

## Screenshots

![01_message close button](https://raw.githubusercontent.com/APTrust/dart/master/docs/accessibility-audit/screenshots/87611/01_message_close_button.png)
